sevntu-checks/src/main/java/com/github/sevntu/checkstyle/checks/coding/ForbidCertainMethodsCheck.java

```diff
@@ -0,0 +1,160 @@
+// Lesser General Public License for more details.
+//
+// You should have received a copy of the GNU Lesser General Public
+// License along with this library; if not, write to the Free Software
+// Foundation, Inc., 59 Temple Place, Suite 330, Boston, MA  02111-1307  USA
+////////////////////////////////////////////////////////////////////////////////
+
+package com.github.sevntu.checkstyle.checks.coding;
+
+import java.util.regex.Pattern;
+
+import com.github.sevntu.checkstyle.Utils;
+import com.puppycrawl.tools.checkstyle.api.AbstractCheck;
+import com.puppycrawl.tools.checkstyle.api.CheckstyleException;
+import com.puppycrawl.tools.checkstyle.api.DetailAST;
+import com.puppycrawl.tools.checkstyle.api.TokenTypes;
+
+/**
+ * Check that forbidden methods are not used. We can forbid methods by ruleName and number of
+ * arguments.
+ * Because of limitations of how checkstyle works we can't add checks on type of the arguments.
+ * This can be used to enforce things like:
+ * <ul>
+ * <li> System.exit() should not be called.</li>
+ * <li> Assert.assertTrue() has a 1 arg variant that does not provide a helpful message on failure.
+ * <li> Thread.sleep() can be prohibited.
+ * </ul>
+ * E.g.:
+ * <p>
+ * {@code
+ * assertTrue(condition);
+ * }
+ * </p>
+ *
+ * @author <a href="mailto:raghavgautam@gmail.com">Raghav Kumar Gautam</a>
+ */
+public class ForbidCertainMethodsCheck extends AbstractCheck {
+
+    /** Key is pointing to the warning message text in "messages.properties" file. */
+    public static final String MSG_KEY_WITH_ARG = "forbid.certain.methods";
+    /** Key is pointing to the warning message text in "messages.properties" file. */
+    public static final String MSG_KEY_WITHOUT_ARG = "forbid.certain.methods.noarg";
+    /** Simple name of this class. */
+    private static final String CLASS_NAME = ForbidCertainMethodsCheck.class.getSimpleName();
+
+    /** Name of the method. */
+    private Pattern methodName;
+
+    /** Regex/String for number of arguments. */
+    private Pattern argumentCount;
+
+    /**
+     * Set method name regex for the forbidden method.
+     * @param methodName regex for the method name
+     * @throws CheckstyleException when methodName is not a valid regex
+     */
+    public void setMethodName(Pattern methodName) throws CheckstyleException {
+        this.methodName = methodName;
+    }
+
+    /**
+     * Set arg count regex for the forbidden method, can be null.
+     * @param argumentCount regex for argument count
+     * @throws CheckstyleException when argumentCount is not a valid regex
+     */
+    public void setArgumentCount(Pattern argumentCount) throws CheckstyleException {
+        this.argumentCount = argumentCount;
+    }
+
+    @Override
+    public void init() {
+        if (methodName == null) {
+            throw new IllegalStateException(String.format(
+                "empty value for methodName (%s) is not allowed in %s",
+                methodName, CLASS_NAME));
```

Comment:
https://travis-ci.org/sevntu-checkstyle/sevntu.checkstyle/jobs/292166801#L1051
Regression is showing this error for the default module which is what it uses. CI must pass for merge.

Don't throw an exception if methodName isn't specified, just have the check do nothing.